python_modules/libraries/dagster-aws/dagster_aws/ecs/tasks.py

```diff
@@ -140,6 +153,44 @@ def task_definition_dict(self):
 
         return kwargs
 
+    def matches_other_task_definition_config(self, other: "DagsterEcsTaskDefinitionConfig"):
+        # doesn't match on every field to ensure that subtle differences in the
+        # fields returned from describe_task_definition and the configuration that
+        # we are specifying here don't result in us always registering a new task
+        # definition
+        if not (
+            self.family == other.family
+            and self.image == other.image
+            and self.container_name == other.container_name
+            and self.command == other.command
+            and self.secrets == other.secrets
+            and self.environment == other.environment
+            and _arns_match(self.execution_role_arn, other.execution_role_arn)
+            and _arns_match(self.task_role_arn, other.task_role_arn)
```

Comment:
reviewer: maybe worth a quick look over what I did *not* include here and verifying that it matches your expectations

e.g. i left out the resource ones (CPU/memory/etc. since those are overridden anyway when you run the task) - but I could add them if we want to be certain